jinahub/indexers/query/vector/FaissIndexer/__init__.py

```diff
@@ -172,5 +265,32 @@ def _load_training_data(self, train_filepath: str) -> 'np.ndarray':
                     result = f.read()
             except Exception as e:
                 self.logger.error(
-                    'Loading training data from binary file failed, filepath={}, {}'.format(train_filepath, e))
+                    'Loading training data from binary file failed, filepath={}, {}'.format(
+                        train_filepath, e
+                    )
+                )
         return result
+
+    def _load_gzip(self, abspath: str, mode='rb') -> Optional['np.ndarray']:
+        try:
+            self.logger.info(f'loading index from {abspath}...')
+            with gzip.open(abspath, mode) as fp:
+                return np.frombuffer(fp.read(), dtype=self.dtype).reshape(
+                    [-1, self.num_dim]
+                )
+        except EOFError:
+            self.logger.error(
+                f'{abspath} is broken/incomplete, perhaps forgot to ".close()" in the last usage?'
+            )
+
+    @property
+    def size(self):
+        if hasattr(self, '_ids'):
+            return len(self._ids)
+        else:
+            return 0
+
+    @requests(on='/fill_embedding')
+    def fill_embedding(self, query_da: DocumentArray, **kwargs):
```

Comment:
this should be called `docs` as per any function of any executor that has an endpoint

Reply:
Good point